pymc_marketing/mmm/base.py

```diff
@@ -390,24 +399,198 @@ def plot_contribution_curves(self) -> plt.Figure:
+
+                polynomial = calculator.get_polynomial()
+                x_space_actual = calculator.get_x_space_actual()
+                y_space_actual = calculator.get_y_space_actual()
+                x_space_projected = calculator.get_x_space_projected()
+                y_space_projected = calculator.get_y_space_projected()
+                roots = calculator.get_roots()
+
+                # Plot the polynomial for the actual data and the projected data
+                ax.plot(
+                    x_space_actual,
+                    y_space_actual,
+                    label=f"{channel} fit",
+                    color=f"C{i}",
+                    linestyle="-",
+                )
+                ax.plot(
+                    x_space_projected, y_space_projected, linestyle="--", color=f"C{i}"
+                )
+
+                # Find the elbow point of the projected curve
+                idx_elbow = find_elbow(x_space_projected, y_space_projected)
+
+                # Get x- and y-values of elbow point
+                elbow_x = x_space_projected[idx_elbow]
+                elbow_y = y_space_projected[idx_elbow]
+
+                for root in roots:
+
+                    if root >= x.min() and root <= x.max():
+                        ax.plot(root, polynomial(root), "ro", label="Plateau Effect")
+                    else:
+                        ax.plot(
+                            root,
+                            polynomial(root),
+                            marker="o",
+                            markersize=8,
+                            markeredgecolor=f"C{i}",
+                            markerfacecolor="white",
+                            label="Projected Plateau Effect",
+                        )
+
+                # Plot the elbow point
+                ax.plot(
+                    elbow_x,
+                    elbow_y,
+                    marker="s",
+                    markersize=8,
+                    markeredgecolor=f"C{i}",
+                    markerfacecolor="white",
+                    label="Optimal Point",
+                )
+
+                # Create a DataFrame for this iteration
+                temp_df = pd.DataFrame(
+                    {
+                        "channel": [channel],
+                        "optimal_point": [elbow_x],
+                        "plateau_point": [root],
+                    }
+                )
+
+                # Use concat to add the new data to the results DataFrame
+                df_estimations = pd.concat([df_estimations, temp_df], ignore_index=True)
+
+            ax.legend(
+                loc="upper left",
+                facecolor="white",
+                title=f"{channel} Legend",
+                fontsize="small",
+            )
+
+            ax.set(xlabel="Spent", ylabel="Contribution")
+        fig.suptitle("Immediate response curves", fontsize=16)
+        return fig, df_estimations
+
+    def budget_allocator(
```

Comment:
I suggest we move this out to an independent module `pymc_marketing/mmm/budget_optimizers`